### PR TITLE
[bitnami/rabbitmq] Disable old metrics, enable new Prometheus auto-scraping

### PR DESCRIPTION
**Pull request by embcla**
This is a fix to allow metrics to work correctly and automatically with prometheus. Enables annotation for automatic metrics discovery, default exporter available since v3.8.0, deprecates pre-v3.8.0 metrics exporter. Chart sub-version bumped up

**Description of the change**
The default now is the following
- the old (pre v.3.8.0) metrics have been disabled
- pods are now annotated for Prometheus to be able to auto-discover and scrape the metrics
- values-production.yaml now has the new prometheus plugin enabled by default
- chart sub-version bumped

**Benefits**

Rabbitmq pods will be automatically found and scraped for metrics by Prometheus

**Possible drawbacks**

unknown

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Comment by rafariossaa** [comment]
Hi, 
  Thank your very much for your PR!!.

  I'm going to open an internal task to:
 - Deprecate [RabbitMQ Exporter](https://github.com/bitnami/bitnami-docker-rabbitmq-exporter) image
 - Adapt [RabbitMQ container](https://github.com/bitnami/bitnami-docker-rabbitmq) to provide a way to configure Prometheus metrics exporters.

We'll keep you updated about our progress. The chart will need changes related to the containers' ones and I, right now, don't know how your PR will fit.

**Comment by andresbono** [comment]
Hi @embcla, we have just updated this PR so it will soon be merged after review. Thanks for your contribution!

File: bitnami/rabbitmq/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: rabbitmq
-version: 6.23.0
+version: 6.24.0
```
Comment (rafariossaa) [inline bitnami/rabbitmq/Chart.yaml:3]: After changing all those settings in the values.yml I will go for a mayor version update.

File: bitnami/rabbitmq/templates/NOTES.txt [hunk 2]
```diff
@@ -60,10 +60,10 @@ To Access the RabbitMQ Management interface:
 
 {{- if .Values.metrics.enabled }}
 
-To access the RabbitMQ Prometheus metrics, get the RabbitMQ Prometheus exporter URL by running:
+To access the RabbitMQ Prometheus metrics, get the RabbitMQ Prometheus URL by running:
 
-    echo "Prometheus Metrics URL: http://127.0.0.1:{{ .Values.metrics.port }}/metrics"
-    kubectl port-forward --namespace {{ .Release.Namespace }} {{ template "rabbitmq.fullname" . }}-0 {{ .Values.metrics.port }}:{{ .Values.metrics.port }}
+    echo "Prometheus Metrics URL: http://127.0.0.1:15692/metrics"
```
Comment (rafariossaa) [inline bitnami/rabbitmq/templates/NOTES.txt:65]: Can we parametrize this port ?

File: bitnami/rabbitmq/templates/NOTES.txt [hunk 3]
```diff
@@ -60,10 +60,10 @@ To Access the RabbitMQ Management interface:
...
+    kubectl port-forward --namespace {{ .Release.Namespace }} {{ template "rabbitmq.fullname" . }}-0 15692:15692
```
Comment (rafariossaa) [inline bitnami/rabbitmq/templates/NOTES.txt:66]: Can we parametrize this port ?

File: bitnami/rabbitmq/templates/networkpolicy.yaml [hunk 4]
```diff
@@ -36,5 +36,5 @@ spec:
 
     # Allow prometheus scrapes
     - ports:
-        - port: {{ .Values.metrics.port }}
+        - port: 15692
```
Comment (rafariossaa) [inline bitnami/rabbitmq/templates/networkpolicy.yaml:39]: Can we parametrize this port ?

File: bitnami/rabbitmq/templates/statefulset.yaml [hunk 5]
```diff
@@ -142,6 +142,10 @@ spec:
           containerPort: {{ .Values.service.distPort }}
         - name: stats
           containerPort: {{ .Values.service.managerPort }}
+        {{- if .Values.metrics.enabled }}
+        - name: metrics
+          containerPort: 15692
```
Comment (rafariossaa) [inline bitnami/rabbitmq/templates/statefulset.yaml:147]: Can we parametrize this port ?

File: bitnami/rabbitmq/templates/svc.yaml [hunk 6]
```diff
@@ -65,7 +60,7 @@ spec:
     {{- end }}
 {{- if .Values.metrics.enabled }}
   - name: metrics
-    port: {{ .Values.metrics.port }}
+    port: 15692
```
Comment (rafariossaa) [inline bitnami/rabbitmq/templates/svc.yaml:63]: Can we parametrize this port ?

File: bitnami/rabbitmq/templates/NOTES.txt (hunk 3 above)
Comment (andresbono) [reply bitnami/rabbitmq/templates/NOTES.txt:66]: It seems it is possible: https://github.com/rabbitmq/rabbitmq-prometheus#configuration

File: bitnami/rabbitmq/Chart.yaml (hunk 1 above)
Comment (andresbono) [reply bitnami/rabbitmq/Chart.yaml:3]: I have doubts about it. The upgrade process seems to go smoothly. Prometheus detects the new endpoint on the fly.

**Comment by andresbono** [comment]
@embcla:

> How's the sidecar supposed to work if, according to my knowledge of RabbitMQ, the metrics are exposed by a plugin that must be loaded by the main executable as an option?

The sidecar container (exporter) is no longer in use.

> You're using the variable _Values.metrics.port_ to configure the service, but I don't see where you're passing this to the sidecar/plugin doing the actual export.

These are the lines passing the metrics port to the rabbitmq prometheus plugin via configuration file:
https://github.com/bitnami/charts/blob/faabbe8b3e8a11b2a18081ed4db3b7ee0493acf0/bitnami/rabbitmq/templates/configuration.yaml#L38-L40

> I can't find the tags of the pod/sidecar to allow Prometheus to automatically find and scrape the target?

This is done here:

https://github.com/bitnami/charts/blob/faabbe8b3e8a11b2a18081ed4db3b7ee0493acf0/bitnami/rabbitmq/templates/statefulset.yaml#L38

`rabbitmq. podAnnotations ` is a new helper function:

https://github.com/bitnami/charts/blob/faabbe8b3e8a11b2a18081ed4db3b7ee0493acf0/bitnami/rabbitmq/templates/_helpers.tpl#L232-L242

I hope this clarifies your doubts.

**Pull request merged**